ytmusicapi/mixins/browsing.py

```diff
@@ -106,10 +109,21 @@ def search(self, query: str, filter: str = None) -> List[Dict]:
                         search_result = self.parser.parse_search_result(data, type)
                         search_results.append(search_result)
 
+                if 'continuations' in res['musicShelfRenderer']:
+                    request_func = lambda additionalParams: self._send_request(
+                        endpoint, body, additionalParams)
+
+                    parse_func = lambda contents: parse_continuation_content_list(
+                        contents, self.parser.parse_search_result)
+
+                    search_results.extend(
+                        get_continuations(res['musicShelfRenderer'], 'musicShelfContinuation',
+                                          limit - len(search_results), request_func, parse_func))
+
         except Exception as e:
             print(str(e))
 
-        return search_results
+        return search_results[:limit]
```

Comment:
The convention in ytmusicapi is to return all retrieved results regardless of the limit, so no need to subset here